### PR TITLE
Parquet writer

### PR DESCRIPTION
**Pull request by acquamarin**
Performance number on ldbc100 comment:
```
kuzu> copy (match (c:Comment) return c.ID, c.locationIP, c.browserUsed, c.content, c.length) to '/tmp/33.parquet';

-
(0 tuples)
Time: 10.81ms (compiling), 52615.40ms (executing)
```

### CODE REVIEW

**Review by ray6080 — APPROVED** [review]
1. In this PR, can we add some benchmark numbers on exporting to parquet file? perhaps on LDBC100 comment table.
2. In following PRs, We should rewrite our testing for converting csv to parquet to use `COPY TO parquet`, which should also increase test coverage for parquet writer.

File: src/include/processor/operator/persistent/reader/parquet/thrift_tools.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 #include <list>
 
+#include "common/file_utils.h"
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/processor/operator/persistent/reader/parquet/thrift_tools.h:4]: This file uses the fileInfo class

File: src/processor/operator/persistent/writer/parquet/parquet_writer.cpp [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    // set up a new row group for this chunk collection
+    auto& row_group = result.rowGroup;
+    row_group.num_rows = ft.getTotalNumFlatTuples();
+    row_group.total_byte_size = row_group.num_rows * ft.getTableSchema()->getNumBytesPerTuple();
+    row_group.__isset.file_offset = true;
+
+    auto& states = result.states;
+    // iterate over each of the columns of the chunk collection and write them
+    assert(ft.getTableSchema()->getNumColumns() == columnWriters.size());
+    std::vector<ColumnWriter*> colWriters;
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/processor/operator/persistent/writer/parquet/parquet_writer.cpp:135]: removed colWriters

File: src/processor/operator/persistent/writer/parquet/parquet_writer.cpp [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    fileMetaData.schema[0].__isset.repetition_type = true;
+
+    std::vector<std::string> schemaPath;
+    for (auto i = 0u; i < this->types.size(); i++) {
+        columnWriters.push_back(ColumnWriter::createWriterRecursive(fileMetaData.schema, *this,
+            this->types[i].get(), this->columnNames[i], schemaPath, mm));
+    }
+}
+
+Type::type ParquetWriter::convertToParquetType(LogicalType* type) {
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/processor/operator/persistent/writer/parquet/parquet_writer.cpp:50]: removed unsupported types

**Pull request merged**